clientv3/maintenance.go

```diff
@@ -145,6 +149,38 @@ func (m *maintenance) Status(ctx context.Context, endpoint string) (*StatusRespo
 	return (*StatusResponse)(resp), nil
 }
 
+func (m *maintenance) Snapshot(ctx context.Context) (io.ReadCloser, error) {
+	ss, err := m.getRemote().Snapshot(ctx, &pb.SnapshotRequest{})
+	if err != nil {
+		return nil, err
+	}
+
+	pr, pw := io.Pipe()
+	go func() {
+		for {
+			resp, err := ss.Recv()
+			if err != nil {
+				pw.CloseWithError(err)
+				return
+			}
+			if resp == nil && err == nil {
+				break
+			}
+			bw := 0
+			for bw < len(resp.Blob) {
+				n, werr := pw.Write(resp.Blob[bw:])
+				if werr != nil {
```

Comment:
https://golang.org/pkg/io/#Writer. If n is smaller than the buffer, there is always an error. so we do not have to track n i guess. go should handle short write retry internally.